### PR TITLE
1834762: crl file candlepin-crl.crl is not found on the candlepin server

### PR DESCRIPTION
**Pull request by Januson**
- Added a check to initialize the CRL file if it is missing during the Candlepin startup

### CODE REVIEW

File: server/src/main/java/org/candlepin/guice/CandlepinContextListener.java [hunk 1]
```diff
@@ -224,6 +229,20 @@ private void initializeSubsystems(Injector injector) throws Exception {
...
+        // Init CRL file
+        this.crlFileUtil = injector.getInstance(CrlFileUtil.class);
+        String filePath = getCrlFilePath();
+        File crlFile = new File(filePath);
+        if (!crlFile.exists() || crlFile.length() == 0) {
+            try {
+                this.crlFileUtil.initializeCRLFile(crlFile, Collections.emptyList());
+            }
+            catch (IOException e) {
+                e.printStackTrace();
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/guice/CandlepinContextListener.java:241]: We should write these to the log and probably rethrow the exception as well.

File: server/src/main/java/org/candlepin/guice/CandlepinContextListener.java [hunk 2]
```diff
@@ -382,4 +401,15 @@ private void insertValidationEventListeners(Injector injector) {
         registry.getEventListenerGroup(EventType.PRE_UPDATE).appendListener(listenerProvider.get());
         registry.getEventListenerGroup(EventType.PRE_DELETE).appendListener(listenerProvider.get());
     }
+
+
+    private String getCrlFilePath() {
+        String filePath = config.getString(ConfigProperties.CRL_FILE_PATH);
+
+        if (filePath == null) {
+            throw new IseException("CRL file path not defined in config file");
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/guice/CandlepinContextListener.java:410]: IseException is a REST exception. We should throw something like a ConfigurationException here instead.

Comment (Januson) [reply server/src/main/java/org/candlepin/guice/CandlepinContextListener.java]: Done.

File: server/src/main/java/org/candlepin/guice/CandlepinContextListener.java (hunk 1 above)
Comment (Januson) [reply server/src/main/java/org/candlepin/guice/CandlepinContextListener.java]: Done.

File: server/src/main/java/org/candlepin/guice/CandlepinContextListener.java [hunk 3]
```diff
@@ -224,6 +228,21 @@ private void initializeSubsystems(Injector injector) throws Exception {
...
+        this.crlFileUtil = injector.getInstance(CrlFileUtil.class);
+        String filePath = getCrlFilePath();
+        File crlFile = new File(filePath);
+        if (!crlFile.exists() || crlFile.length() == 0) {
+            try {
+                this.crlFileUtil.initializeCRLFile(crlFile, Collections.emptyList());
+            }
+            catch (IOException e) {
+                log.error("Error occurred during initialization of CRL file!", e);
+                throw new ConfigurationException("Error occurred during initialization of CRL file!", e);
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/guice/CandlepinContextListener.java:241]: This one should stay as an IOException, just rethrown.

Comment (Januson) [reply server/src/main/java/org/candlepin/guice/CandlepinContextListener.java]: Done.

**Pull request merged**